src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/hotshard/HighCPUShardRca.java

```diff
@@ -0,0 +1,209 @@
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.contexts.ResourceContext;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.flow_units.MetricFlowUnit;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.flow_units.ResourceFlowUnit;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.HotNodeSummary;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.HotShardSummary;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.scheduler.FlowUnitOperationArgWrapper;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.reader.ClusterDetailsEventProcessor;
+import java.time.Clock;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jooq.Record;
+
+/**
+ * This RCA is to identify a hot shard within an index. A Hot shard is an outlier within its counterparts.
+ * The RCA subscribes to following metrics :
+ * 1. CPU_Utilization
+ * 2. IO_TotThroughput
+ * 3. IO_TotalSyscallRate
+ *
+ * <p>The RCA looks at the above 3 metric data, compares the values against the threshold for each resource
+ * and if the usage for any of 3 resources is greater than their individual threshold, we mark the context
+ * as 'UnHealthy' and create a HotShardResourceSummary for the shard.
+ *
+ * <p>Optional metrics which can be added in future :
+ * 1. Heap_AllocRate
+ * 2. Paging_RSS
+ *
+ */
+public class HighCPUShardRca extends Rca<ResourceFlowUnit> {
+
+    private static final Logger LOG = LogManager.getLogger(HighCPUShardRca.class);
+    private static final int SLIDING_WINDOW_IN_SECONDS =  60;
+
+    //TODO : {@khushbr} refine the threshold values and read same from config file
+    private static final double CPU_USAGE_THRESHOLD = 0.01;
+    private static final double IO_TOT_THROUGHPUT_THRESHOLD_IN_BYTES = 250000;
+    private static final double IO_TOT_SYSCALL_RATE_THRESHOLD_PER_SECOND = 0.01;
+
+    private final Metric cpuUsage;
+    private final Metric ioTotThroughput;
+    private final Metric ioTotSyscallRate;
+    private final ResourceType resourceType;
+    private final int rcaPeriod;
+    private int counter;
+    protected Clock clock;
+
+    // HashMap with IndexShardKey object as key and SlidingWindowData object of metric data as value
+    private HashMap<IndexShardKey, SlidingWindow<SlidingWindowData>> cpuUsageHashMap;
+    private HashMap<IndexShardKey, SlidingWindow<SlidingWindowData>> ioTotThroughputHashMap;
+    private HashMap<IndexShardKey, SlidingWindow<SlidingWindowData>>  ioTotSyscallRateHashMap;
+
+    public <M extends Metric> HighCPUShardRca(final long evaluationIntervalSeconds, final int rcaPeriod,
```

Comment:
Similarly here and remove the generic parameter

Reply:
Done. Updated.